src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneCaptureMethodSender.php

```diff
@@ -0,0 +1,164 @@
+        OrderPriceDistributorInterface $orderPriceDistributor,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PayoneRequestProductDataMapperInterface $payoneRequestProductDataMapper,
+        OrderHandlingMapperInterface $orderHandlingMapper
+    ) {
+        parent::__construct($queryContainer, $paymentMapperReader);
+        $this->executionAdapter = $executionAdapter;
+        $this->standardParameter = $standardParameter;
+        $this->orderPriceDistributor = $orderPriceDistributor;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->payoneRequestProductDataMapper = $payoneRequestProductDataMapper;
+        $this->orderHandlingMapper = $orderHandlingMapper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneCaptureTransfer $captureTransfer
+     *
+     * @return \Generated\Shared\Transfer\CaptureResponseTransfer
+     */
+    public function capturePayment(PayoneCaptureTransfer $captureTransfer): CaptureResponseTransfer
+    {
+        $distributedPriceOrderTransfer = $this->orderPriceDistributor->distributeOrderPrice(
+            $captureTransfer->getOrder()
+        );
+        $captureTransfer->setOrder($distributedPriceOrderTransfer);
+        $captureTransfer->setAmount($distributedPriceOrderTransfer->getTotals()->getGrandTotal());
+
+        $paymentEntity = $this->getPaymentEntity($captureTransfer->getPayment()->getFkSalesOrder());
+        $paymentMethodMapper = $this->getPaymentMethodMapper($paymentEntity);
+
+        $requestContainer = $paymentMethodMapper->mapPaymentToCapture($paymentEntity);
+
+        if ($captureTransfer->getAmount()) {
+            $requestContainer = $this->payoneRequestProductDataMapper->mapData($captureTransfer->getOrder(), $requestContainer);
+            $requestContainer = $this->orderHandlingMapper->prepareOrderHandling($captureTransfer->getOrder(), $requestContainer);
+        }
+
+        /** @var \SprykerEco\Zed\Payone\Business\Api\Request\Container\CaptureContainer $requestContainer */
+        if (!empty($captureTransfer->getSettleaccount())) {
+            $businnessContainer = new BusinessContainer();
+            $businnessContainer->setSettleAccount($captureTransfer->getSettleaccount());
+            $requestContainer->setBusiness($businnessContainer);
+        }
+
+        if ($captureTransfer->getAmount() !== null) {
+            $requestContainer->setAmount($captureTransfer->getAmount());
+        }
+
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+
+        $apiLogEntity = $this->initializeApiLog($paymentEntity, $requestContainer);
+        $responseContainer = new CaptureResponseContainer($rawResponse);
+        $this->updateApiLogAfterCapture($apiLogEntity, $responseContainer);
+
+        $responseMapper = new CaptureResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
++